app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +138,21 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit(phone)
+      @phone = Pii::Fingerprinter.fingerprint(Phonelib.parse(phone).e164)
+      submission_rate_limiter ||= RateLimiter.new(
+        target: @phone,
+        rate_limit_type: :phone_submissions,
+      )
+      submission_rate_limiter.increment!
+
+      lock_out_user if submission_rate_limiter.maxed?
```

Comment:
I think this could be potentially problematic in the case where I maliciously submit the same number a bunch of times, and then you come along within the next N timeframe and submit it for the first time (that you're aware of), and the rate limiter locks you out.